### PR TITLE
[widget] disable bsp while opening non BSP project

### PR DESCRIPTION
**Pull request by azdrojowa123**

### CODE REVIEW

File: src/main/kotlin/org/jetbrains/plugins/bsp/import/BspInitializer.kt [hunk 1]
```diff
@@ -23,39 +25,48 @@ import org.jetbrains.plugins.bsp.ui.widgets.toolwindow.all.targets.BspAllTargets
  */
 public class BspInitializer : StartupActivity {
   override fun runActivity(project: Project) {
-    println("BspInitializer.runActivity")
-
     val connectionService = project.getService(BspConnectionService::class.java)
-    val magicMetaModelService = MagicMetaModelService.getInstance(project)
 
-    val task = object : Task.Backgroundable(project, "Loading changes...", true) {
+    if (connectionService.isRunning()) {
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/import/BspInitializer.kt:30]:
```suggestion
    if (project.isNewProject()) {
```

File: src/main/kotlin/org/jetbrains/plugins/bsp/import/BspProjectOpenProcessor.kt [hunk 2]
```diff
@@ -28,13 +28,17 @@ public class BspProjectOpenProcessor : ProjectOpenProcessor() {
...
   override fun getIcon(): Icon = BspPluginIcons.bsp
 
-  override fun canOpenProject(file: VirtualFile): Boolean {
-    val bspConnectionFilesProvider = BspConnectionFilesProvider(file)
-    val bspConnectionDetailsGeneratorProvider =
-      BspConnectionDetailsGeneratorProvider(file, BspConnectionDetailsGeneratorExtension.extensions())
+  override fun canOpenProject(file: VirtualFile): Boolean = BspProjectOpenProcessor.canOpenProject(file)
+
+  public companion object {
+    public fun canOpenProject(file: VirtualFile): Boolean {
```
Comment (abrams27) [inline src/main/kotlin/org/jetbrains/plugins/bsp/import/BspProjectOpenProcessor.kt:34]: do we need it here?

Comment (azdrojowa123) [reply src/main/kotlin/org/jetbrains/plugins/bsp/import/BspProjectOpenProcessor.kt:34]: no - I need it for another (openProvider) purpose and forgot to remove

File: src/main/kotlin/org/jetbrains/plugins/bsp/import/BspInitializer.kt (hunk 1 above)
Comment (azdrojowa123) [reply src/main/kotlin/org/jetbrains/plugins/bsp/import/BspInitializer.kt:30]: hmm but when I open e.g. bazel-bsp via maven it still go into this if statement with this change - and we don't want this behavior do we?

**Review by abrams27 — APPROVED** [review]

**Pull request merged**